src/NServiceBus.Core/Utils/Reflection/Reflect.cs

```diff
@@ -30,7 +30,7 @@ public static PropertyInfo GetProperty(Expression<Func<TTarget, object>> propert
             return GetMemberInfo(property, checkForSingleDot) as PropertyInfo;
         }
 
-        static MemberInfo GetMemberInfo(Expression member, bool checkForSingleDot)
+        public static MemberInfo GetMemberInfo(Expression member, bool checkForSingleDot)
```

Comment:
@danielmarbach If I only have access to GetProperty and not GetMemberInfo, then all I know from SagaMetadata is that the mapping was unsuccessful, and I won't have the the MemberInfo so I can tell the user the name of the saga property that's at fault in the exception.

The class itself is internal. I can make the method internal as well but it won't change anything.